src/components/input/input.ts

```diff
@@ -480,7 +500,7 @@ export default class SlInput extends ShoelaceElement implements ShoelaceFormCont
               inputmode=${ifDefined(this.inputmode)}
               aria-describedby="help-text"
               @change=${this.handleChange}
-              @input=${this.handleInput}
+              @input=${this.currency === 'usd' ? this.handleCurrencyInput : this.handleInput}
```

Comment:
maybe this check can happen in `this.handleInput`?
I think we'll probably need the `this.emit('sl-change')` too although im not exactly sure what it does

Reply:
I would definitely prefer to avoid overriding the default handler just for this use case, which could cause maintenance problems down the road if changes are made to the normal input handler. If we need it, this check can happen within `handleInput`, as Anne mentioned.